### PR TITLE
Simplify some proofs in Categories.Morphism.Isomorphism

### PR DESCRIPTION
**Pull request by rei1024**
- Added properties of groupoid to Categories.Category.Groupoid.Properties
- ~~Removed duplicated module definitions in Categories.Category.Construction.Kleisli~~

### CODE REVIEW

File: Categories/Morphism/Isomorphism.agda [hunk 1]
```diff
@@ -188,10 +50,7 @@ module _ where
   --
   flip-iso : (f : A ≅ B) {g : A ⇒ C} {h : B ⇒ C} →
              g ≈ h ∘ from f → g ∘ to f ≈ h
-  flip-iso f {g} {h} tr₁ = begin
-    g ∘ to f            ≈⟨ pushˡ tr₁ ⟩
-    h ∘ from f ∘ to f   ≈⟨ elimʳ (isoʳ f) ⟩
-    h                   ∎
+  flip-iso f tr₁ = sym (switch-fromtoʳ f (sym tr₁))
```
Comment (sstucki) [inline Categories/Morphism/Isomorphism.agda]: Hm, this makes me think I should have defined `flip-iso` and `push-eq` in `Categories.Morphism.Reasoning.Iso` instead. Would you mind moving them there and updating the dependencies? Or I could do that in a follow-up PR.

File: Categories/Morphism/Isomorphism.agda [hunk 2]
```diff
@@ -46,132 +34,6 @@ open MCore using () renaming (_∘_ to _∘ᵢ_) public
...
-    lift (_ ∼⁺⟨ [ g′ ] ⟩ [ f′ ]) ≃⁺ lift (_ ∼⁺⟨ [ i′ ] ⟩ [ h′ ])
-  lift-square eq f′ g′ h′ i′ = lift-cong (_ ∼⁺⟨ [ g′ ] ⟩ [ f′ ]) (_ ∼⁺⟨ [ i′ ] ⟩ [ h′ ]) eq
-
-  lift-pentagon : {f : A ⇒ B} {g : C ⇒ A} {h : D ⇒ C} {i : E ⇒ B} {j : D ⇒ E} {l : E ⇒ D}
-                  {a : B ⇒ A} {b : A ⇒ C} {c : C ⇒ D} {d : B ⇒ E} →
-    f ∘ g ∘ h ≈ i ∘ j →
-    (f′ : Iso f a) → (g′ : Iso g b) → (h′ : Iso h c) → (i′ : Iso i d) → (j′ : Iso j l) →
-    lift (_ ∼⁺⟨ _ ∼⁺⟨ [ h′ ] ⟩ [ g′ ] ⟩ [ f′ ]) ≃⁺ lift (_ ∼⁺⟨ [ j′ ] ⟩ [ i′ ])
-  lift-pentagon eq f′ g′ h′ i′ j′ = lift-cong (_ ∼⁺⟨ _ ∼⁺⟨ [ h′ ] ⟩ [ g′ ] ⟩ [ f′ ]) (_ ∼⁺⟨ [ j′ ] ⟩ [ i′ ]) eq
-
```
Comment (sstucki) [inline Categories/Morphism/Isomorphism.agda]: I'm wondering whether all of this stuff about transitive closures should really be removed. I have the feeling that it was meant to be generic functionality that could be useful in proofs other than the ones below. Would like to hear @HuStmpHrrr's opinion about this (I think he added much of this in 46f79440).

**Comment by sstucki** [comment]
The changes in `Categories/Category/Construction/Kleisli.agda` should probably go into a separate commit.

File: Categories/Morphism/Isomorphism.agda (hunk 1 above)
Comment (rei1024) [reply Categories/Morphism/Isomorphism.agda]: OK. I'll do that.

**Comment by JacquesCarette** [comment]
Could you further split this?  The addition of `Categories.Category.Groupoid.Properties` is ready to go, and I would merge that in.  There is a lot of merit to the other pieces too, but they need a little bit of discussion.

In general, try to keep PRs as small and self-contained as possible.

**Comment by rei1024** [comment]
Ok. I will close this and split into multiple PRs.

**Pull request closed without merging**